dvc/command/data_sync.py

```diff
@@ -37,6 +37,7 @@ def run(self):
                 force=self.args.force,
                 recursive=self.args.recursive,
                 run_cache=self.args.run_cache,
+                glob=self.args.glob,
```

Comment:
Looks like we are also missing it in fetch, but that will require some additional modification, so out of scope for this PR.